<launch>

  <!-- the camera -->

  <!-- <include file="$(find azure_kinect_ros_driver)/launch/kinect_rgbd.launch">
    <arg name="depth_enabled" value="true" />
    <arg name="depth_mode" value="WFOV_UNBINNED" />
    <arg name="fps" value="15" />
    <arg name="color_resolution" value="1536P" />
    <arg name="num_worker_threads" value="4" />
    <arg name="overwrite_robot_description" value="false" />
  </include> -->

  <include file="$(find realsense2_camera)/launch/rs_camera.launch" />

  <!-- the Apriltap detection -->

  <include file="$(find leap_task_A)/launch/tag_detect.launch"></include>

  <!-- publish the transform between camera and hand -->

  <arg name="tracking_base_frame" default="camera_link" />
  <arg name="calibration_file" default="$(find leap_hardware)/config/rgmc_d405_calib.yaml" />

  <include file="$(find easy_handeye)/launch/publish.launch">
    <arg name="eye_on_hand" value="false" />
    <arg name="namespace_prefix" value="leap_kinect_calib" />
    <arg name="tracking_base_frame" value="$(arg tracking_base_frame)" />
    <arg name="calibration_file" value="$(arg calibration_file)" />
  </include>

  <!-- Tag TF to a pose topic -->

  <node pkg="leap_task_A" name="tag_tf_to_topic" type="tag_tf_to_topic.py" output="screen" />
  <node pkg="leap_task_A" name="task_goal_transform" type="task_goal_transform.py"
    output="screen" />

  <!-- Leap Hand -->

  <include file="$(find leap_hardware)/launch/leap_bringup.launch">
    <arg name="free_move" value="false" />
    <arg name="enable_visualization" value="true" />
    <arg name="publish_joint_state" value="false" />
    <arg name="use_rviz" value="false" />
  </include>

  <!-- <node pkg="leap_hardware" name="leaphand_publisher" type="leap_joint_state_publisher.py"
    output="screen" /> -->

  <!-- rviz -->

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find leap_task_A)/rviz/taskA.rviz" />

</launch>